query/src/main/scala/filodb/query/exec/SelectRawPartitionsExec.scala

```diff
@@ -38,18 +39,39 @@ final case class SelectRawPartitionsExec(id: String,
                                          colIds: Seq[Types.ColumnId]) extends LeafExecPlan {
   import SelectRawPartitionsExec._
 
-  require(colIds.nonEmpty)
-
   protected[filodb] def schemaOfDoExecute(dataset: Dataset): ResultSchema = {
-    val numRowKeyCols = colIds.zip(dataset.rowKeyIDs).takeWhile { case (a, b) => a == b }.length
+    require(dataset.rowKeyIDs.forall(rk => !colIds.contains(rk)),
+      "User selected columns should not include timestamp (row-key); it will be auto-prepended")
```

Comment:
Not sure we need to check for this, or what this really has to do with this PR.  Also this code won't work post-schema as we got rid of rowKeyIDs.  Please just leave it out to make merges easier.

Reply:
As mentioned in another comment, I want one place where column selection happens. The require makes the logic easy to understand.